src/Symfony/Component/Messenger/Stamp/AckStamp.php

```diff
@@ -21,7 +21,7 @@ final class AckStamp implements NonSendableStampInterface
     private $ack;
 
     /**
-     * @param \Closure(Envelope, \Throwable|null) $ack
+     * @param \Closure(Envelope, \Throwable|null): void $ack
```

Comment:
The return value is also ignored here.

Reply:
which is exactly what void conveys

Reply:
The `void` return type ensure the function doesn't return anything; PHP triggers a fatal error if it does. Here, we simply ignore what is returned it the function returns something.

A developer can implement a function that return something (for test purposes for example), changing the type here would be a breaking change.

Reply:
Actually, PhpStorm static analysis+highlight of PHPdoc callable only works when a return type is provided to the callable or closure. Maybe let's use the mixed return type?

Reply:
> changing the type here would be a breaking change

it's a soft type, and could trigger a deprecation similary as we did with method return types. But that's a different story :')

I agree no type/mixed improves general compability for a closure, but im not a big fan of it. I tend to like to know return value is irrelevant.

Reply:
I agree adding "mixed" could be confusing when reading the phpdoc, where no type at all "doesn't raise any question". 🤔